elements/lisk-transactions/src/14_unlock_transaction.ts

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { intToBuffer } from '@liskhq/lisk-cryptography';
+import { isNumberString, validator } from '@liskhq/lisk-validator';
+
+import {
+	BaseTransaction,
+	StateStore,
+	StateStorePrepare,
+} from './base_transaction';
+import { convertToAssetError, TransactionError } from './errors';
+import { Account, TransactionJSON } from './transaction_types';
+import { sortUnlocking } from './utils';
+
+export interface Unlock {
+	readonly delegateAddress: string;
+	readonly amount: bigint;
+	readonly unvoteHeight: number;
+}
+
+export interface UnlockAsset {
+	readonly unlockingObjects: ReadonlyArray<Unlock>;
+}
+
+const voteAssetFormatSchema = {
+	type: 'object',
+	required: ['unlockingObjects'],
+	properties: {
+		unlockingObjects: {
+			type: 'array',
+			minItems: 1,
+			maxItems: 20,
+			items: {
+				type: 'object',
+				required: ['delegateAddress', 'amount', 'unvoteHeight'],
+				properties: {
+					delegateAddress: {
+						type: 'string',
+						format: 'address',
+					},
+					amount: {
+						type: 'string',
+						format: 'int64',
+					},
+					unvoteHeight: {
+						type: 'integer',
+						minimum: 0,
+					},
+				},
+			},
+		},
+	},
+};
+
+const SIZE_UINT32 = 4;
+const SIZE_INT64 = 8;
+const SIZE_UINT64 = SIZE_INT64;
+// tslint:disable-next-line no-magic-numbers
+const TEN_UNIT = BigInt(10) * BigInt(10) ** BigInt(8);
```

Comment:
I don't know its a bit unclear with the constant name, maybe something like `SELF_VOTE_AMOUNT_MULTIPLIER`

Reply:
This is the `10 LSK` which all the vote needs to be multiply of but i wanted to avoid using LSK in the SDK.

Reply:
Yes its 10 LSK, but I was thinking that this is a constant that is used in the context of self votes and maybe in future if we will have to change it then we don't have to change the constant name but just the value (although the chances are that it will never be changed)

Reply:
But not a very important comment, if there is a better name then we can use it otherwise we can keep it as it is

Reply:
I also think this name is not clear, any suggestion? This constant is used in self vote and normal vote

Reply:
`AMOUNT_MULTIPLIER_FOR_VOTES` or `TEN_TIMES_AMOUNT_FOR_VOTES`? these are the only ones I can think of right now, but if you have any better name then feel free to use it